### PR TITLE
Add type parameter to remaining Series.astype overloads.

### PR DESCRIPTION
**Pull request by JanEricNitschke**
- [x] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

Closes https://github.com/pandas-dev/pandas-stubs/issues/948

Verified that some of the test statements [here](https://github.com/pandas-dev/pandas-stubs/blob/main/tests/test_series.py#L2191) pass pyright strict with this change that do not do so without.

Tested using `object` instead of `Any` as that is the runtime behaviour, but that did not work due to the parameter bounds.

Other option (maybe better?) would be to explicitely return as series matching the specified parameter? But i think that does not work either because the [bounds of the type parameter for series](https://github.com/pandas-dev/pandas-stubs/blob/main/pandas-stubs/_typing.pyi#L533) are more restrictive than the input arguments for `astype`.

### CODE REVIEW

File: pandas-stubs/core/series.pyi [hunk 1]
```diff
@@ -1320,14 +1320,14 @@ class Series(IndexOpsMixin[S1], NDFrame):
         dtype: CategoryDtypeArg,
         copy: _bool = ...,
         errors: IgnoreRaise = ...,
-    ) -> Series: ...
+    ) -> Series[Any]: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/series.pyi:1323]:
```suggestion
    ) -> Series[CategoricalDtype]: ...
```
If this change is failing a test, then modify the test.

File: pandas-stubs/core/series.pyi [hunk 2]
```diff
@@ -1320,14 +1320,14 @@ class Series(IndexOpsMixin[S1], NDFrame):
...
     @overload
     def astype(
         self,
         dtype: ObjectDtypeArg | VoidDtypeArg | ExtensionDtype | DtypeObj,
         copy: _bool = ...,
         errors: IgnoreRaise = ...,
-    ) -> Series: ...
+    ) -> Series[Any]: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/series.pyi:1330]: This is correct.  Any generic parameter to `Series` needs to be in the `TypeVar` `S1` that is declared in `_typing.pyi`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: pandas-stubs/core/series.pyi (hunk 1 above)
Comment (JanEricNitschke) [reply pandas-stubs/core/series.pyi]: Made that change and made one of the test that previously expected the `Series[Any]` fit with this.

Also double checked that pyright strict that not throw any new warnings due to this.

File: pandas-stubs/core/series.pyi [hunk 3]
```diff
@@ -1320,14 +1320,14 @@ class Series(IndexOpsMixin[S1], NDFrame):
         dtype: CategoryDtypeArg,
         copy: _bool = ...,
         errors: IgnoreRaise = ...,
-    ) -> Series: ...
+    ) -> Series[CategoryDtypeArg]: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/series.pyi:1323]: This needs to be `CategoryDtype` as it is value in `S1`

File: tests/test_series.py [hunk 4]
```diff
@@ -2581,8 +2581,8 @@ def test_astype_categorical(cast_arg: CategoryDtypeArg, target_type: type) -> No
 
     if TYPE_CHECKING:
         # pandas category
-        assert_type(s.astype(pd.CategoricalDtype()), "pd.Series[Any]")
-        assert_type(s.astype("category"), "pd.Series[Any]")
+        assert_type(s.astype(pd.CategoricalDtype()), "pd.Series[CategoryDtypeArg]")
+        assert_type(s.astype("category"), "pd.Series[CategoryDtypeArg]")
```
Comment (Dr-Irv) [inline tests/test_series.py:2585]: Should be `CategoryDtype`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: pandas-stubs/core/series.pyi (hunk 3 above)
Comment (JanEricNitschke) [reply pandas-stubs/core/series.pyi]: Made the change

File: tests/test_series.py (hunk 4 above)
Comment (JanEricNitschke) [reply tests/test_series.py]: Here too

**Review by Dr-Irv — APPROVED** [review]
thanks @JanEricNitschke

**Pull request merged**